test/ec_internal_test.c

```diff
@@ -156,6 +156,55 @@ static int field_tests_ecp_mont(void)
 }
 
 #ifndef OPENSSL_NO_EC2M
+/* Test that decoding of invalid GF2m field parameters fails. */
+static int ec2m_field_sanity(void)
+{
+    int ret = 0;
+    BN_CTX *ctx = BN_CTX_new();
+    BIGNUM *p, *a, *b;
+    EC_GROUP *group1 = NULL, *group2 = NULL, *group3 = NULL;
+
+    TEST_info("Testing GF2m hardening\n");
+
+    BN_CTX_start(ctx);
+    p = BN_CTX_get(ctx);
+    a = BN_CTX_get(ctx);
+    if (!TEST_ptr(b = BN_CTX_get(ctx))
+        || !TEST_true(BN_one(a))
+        || !TEST_true(BN_one(b)))
+        goto out;
+
+    /* Even pentanomial value should be rejected */
+    if (!TEST_true(BN_set_word(p, 0xf2)))
+        goto out;
+    if (!TEST_ptr_null(group1 = EC_GROUP_new_curve_GF2m(p, a, b, ctx)))
+        TEST_error("Zero constant term accepted in GF2m polynomial");
+
+    /* Odd hexanomial should also be rejected */
+    BN_set_word(p, 0xf3);
```

Comment:
You've tested the return of `BN_set_word` above but not this one?

Reply:
Surely once `p` holds a previous "word", there's room to change it without any new allocation, so the update can't fail?

Reply:
Well, the docs don't limit the reasons why something might fail. It just says that function returns 1 on success or 0 on failure. Looking "under the covers" we can discover the reasons for failure and your rationale holds up - until someone changes the code and some other reason for failure is introduced. So as a matter of general principle I don't think we should "assume" the reasons why a function might fail or otherwise.